siege_utilities/schema/migration_runner.py

```
@@ -0,0 +1,358 @@
+                break
+        return applied
+
+    # ------------------------------------------------------------------
+    # Internals
+    # ------------------------------------------------------------------
+
+    def _apply_one(self, migration: PendingMigration) -> None:
+        """Apply a single migration + record it in the tracking table in one transaction."""
+        sql_text = migration.path.read_text(encoding="utf-8")
+        with psycopg.connect(self._dsn) as conn:
+            with conn.transaction():
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)